### PR TITLE
[Feature][Plugin] Add plugin-spark-source-kudu into v2

### PR DESCRIPTION
**Pull request by zixi0825**
---

## Purpose of this pull request

Add plugin-spark-source-kudu related: #744

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [x] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

File: plugin-spark-source-kudu/src/main/scala/io/github/interestinglab/waterdrop/spark/source/Kudu.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+package io.github.interestinglab.waterdrop.spark.source
+
+import io.github.interestinglab.waterdrop.spark.SparkEnvironment
+import org.apache.kudu.spark.kudu._
+import io.github.interestinglab.waterdrop.spark.batch.SparkBatchSource
+import org.apache.spark.sql.{Dataset, Row, SparkSession}
+
+class Kudu extends SparkBatchSource {
+
+  override def checkConfig(): (Boolean, String) = {
```
Comment (simon824) [inline plugin-spark-source-kudu/src/main/scala/io/github/interestinglab/waterdrop/spark/source/Kudu.scala:26]: Need to return CheckResult

File: plugin-spark-source-kudu/pom.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-streaming_2.11</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.kudu</groupId>
+            <artifactId>kudu-spark2_2.11</artifactId>
+            <version>1.7.0</version>
```
Comment (CalvinKirs) [inline plugin-spark-source-kudu/pom.xml:40]: Unified management version under the root pom

**Review by simon824 — APPROVED** [review]
LGTM

**Comment by CalvinKirs** [comment]
please add docs to https://github.com/InterestingLab/seatunnel/tree/dev/docs/en/configuration/source-plugins
thanks

**Review by wuchunfu — COMMENTED** [review]
<build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-deploy-plugin</artifactId>
            </plugin>

            
            
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-source-plugin</artifactId>
            </plugin>

            <plugin>
                <groupId>org.sonatype.plugins</groupId>
                <artifactId>nexus-staging-maven-plugin</artifactId>
            </plugin>

            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-gpg-plugin</artifactId>
            </plugin>

            <plugin>
                <groupId>net.alchim31.maven</groupId>
                <artifactId>scala-maven-plugin</artifactId>
            </plugin>
        </plugins>
    </build>

**Comment by wuchunfu** [comment]
Here is no longer needed, can you update the upstream code.

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**